internal/app/app.go

```diff
@@ -26,17 +27,18 @@ import (
 
 // App is main application structure
 type App struct {
-	state              appState
-	logger             *log.Logger
-	config             *config.Config
-	dcs                dcs.DCS
-	cluster            *mysql.Cluster
-	filelock           *flock.Flock
-	nodeFailedAt       map[string]time.Time
-	streamFromFailedAt map[string]time.Time
-	daemonState        *DaemonState
-	daemonMutex        sync.Mutex
-	replRepairState    map[string]*ReplicationRepairState
+	state               appState
+	logger              *log.Logger
+	config              *config.Config
+	dcs                 dcs.DCS
+	cluster             *mysql.Cluster
+	filelock            *flock.Flock
+	nodeFailedAt        map[string]time.Time
+	streamFromFailedAt  map[string]time.Time
+	daemonState         *DaemonState
+	daemonMutex         sync.Mutex
+	replRepairState     map[string]*ReplicationRepairState
+	ExternalReplication replication.IExternalReplication
```

Comment:
Maybe it's better not to add this field to the app and add it to the node

Reply:
I want to make "strategy" pattern for `IExternalReplication` in `app`. In this case I think we can't use `Node` to reach purpose.